spec/components/previews/persona_sign_in_component_preview.rb

```diff
@@ -0,0 +1,27 @@
+class PersonaSignInComponentPreview < ApplicationComponentPreview
+  def user_anne
+    persona = FactoryBot.build(:placements_support_user, :anne)
+    render_with_template(template:, locals: { persona: })
+  end
+
+  def user_mary
+    persona = FactoryBot.build(:placements_user, :mary)
+    render_with_template(template:, locals: { persona: })
+  end
+
+  def user_patricia
+    persona = FactoryBot.build(:placements_user, :patricia)
+    render_with_template(template:, locals: { persona: })
+  end
+
+  def support_user_colin
+    persona = FactoryBot.build(:placements_support_user, :colin)
+    render_with_template(template:, locals: { persona: })
+  end
+
+  private
+
+  def template
+    "templates/persona_sign_in_component_preview"
+  end
```

Comment:
Would it be more simple / less code to follow the default directory?

https://viewcomponent.org/guide/previews.html#preview-templates

> Given a preview test/components/previews/cell_component_preview.rb, template files can be defined at test/components/previews/cell_component_preview/:

Reply:
So for that example it works because the method is `default` and the template is `test/components/previews/cell_component_preview/default.html.erb`.
Which means you need a different template per-method.
In my implementation, the templates are stored in a `templates` folder, and you only need to define the one template (i.e `templates/persona_sign_in_component_preview`).
This is only necessary in this case because for the component to look correct it needs some additional html. This will not be the case for every component preview.

Reply:
I see what you mean. We'd need a copy of our template _per_ preview.

Hmm, initially, it feels like we're not _directly_ interacting with the component for the previews.

But it doesn't really matter either.